kernels/optimized/cpu/op_linear.cpp

```diff
@@ -50,8 +47,9 @@ Tensor& opt_linear_out(
   for (int ii = 0; ii < in.dim() - 1; ++ii) {
     flattened_input_dim *= in.sizes()[ii];
   }
+  static constexpr const char op_name[] = "linear.out";
   ET_SWITCH_REAL_TYPES_AND2(
-      Half, BFloat16, in.scalar_type(), ctx, "mm.out", CTYPE, [&]() {
```

Comment:
Hmm, this is tricky. linear isn't a core ATen op so I  *think* we aren't supposed to do this; instead I *think* if there is a bias it's addmm.out and if not it's mm.out. This matters for selective build configurations and depends on how XNNPACK does it, so CC @kimishpatel / @digantdesai / @mcr229  ?

Reply:
@larryliu0820 can you say something about how this impacts selective build? Are we relying on mm.out to apply selective dtype rule for all instances of op_mm?
@swolchok are you also saying that dtype selective build looks at op_mm in the edge ir graph and then apply selective build rule that match "mm.out" in the op name string?

Reply:
I think this is a right fix. For linear op we should use "linear.out" here. Selective build will have the following impact with this change: before this change, if a user select a dtype (say float16) for `mm.out`, it will apply to linear.out *by mistake*, like a side effect. After this change, any binary that relies this side effect will have a small binary size regression, because the key changes from "mm.out" to "linear.out". Though I would argue that this is still a right fix and we should fix whatever binary size regression by properly selecting "linear.out".

Reply:
Thanks @larryliu0820 that was my understanding too. But @hsharma35 can we please use the other diff as the right fix. I dont think we should really land this as is if we have a better fix available. If this is urgent for you, lets prioritze the other one